components/selection/selection-header.js

```diff
@@ -76,14 +76,13 @@ export class SelectionHeader extends SelectionObserverMixin(RtlMixin(LocalizeCor
 			.d2l-selection-header-container-slim {
 				min-height: 36px;
 			}
-			d2l-selection-select-all {
+			d2l-selection-select-all, d2l-selection-summary {
 				flex: none;
 			}
-			d2l-selection-summary {
-				flex: none;
+			d2l-selection-select-all + d2l-selection-summary {
 				margin-left: 0.9rem;
 			}
-			:host([dir="rtl"]) d2l-selection-summary {
+			:host([dir="rtl"]) d2l-selection-select-all + d2l-selection-summary {
 				margin-left: 0;
 				margin-right: 0.9rem;
 			}
```

Comment:
These styles were tightened so that the spacing is still correct when the select-all is omitted (for the table-header).